kliff/trainer/option_enumerations.py

```diff
@@ -0,0 +1,225 @@
+from enum import Enum
+
+from kliff._exceptions import TrainerError
+
+
+class ModelTypes(Enum):
+    """
+    Enumerates the different types of models that can be used in the training
+    process. The different types are:
+    - KIM: The KIM model, passed as a string of KIM ID.
+    - TORCH: A model implemented in PyTorch, passed as a Python callable.
+    - TAR: A model saved in a tar file, which can either be a valid KIM API
+        model, where it will be installed in CWD using kim-api collections
+        management, or TorchML driver based model, in which case, the TorchScript
+        file will be extracted and used. The kind of model is determined from the
+        KIM API CMakelists.txt file model drive name string.
+    """
+
+    UNDEFINED = -1
+    KIM = 0
+    TORCH = 1
+    TAR = 2
+
+    @staticmethod
+    def get_model_enum(input_str: str):
+        """
+        Get the model type from the input string.
+
+        Args:
+            input_str: name of the model type. "kim", "torch", "pt", "pth" or "tar"
+
+        Returns:
+            Model type enum.
+        """
+        if input_str.lower() == "kim":
+            return ModelTypes.KIM
+        elif (
+            input_str.lower() == "torch"
+            or input_str.lower() == "pt"
+            or input_str.lower() == "pth"
+        ):
+            return ModelTypes.TORCH
+        elif input_str.lower() == "tar":
+            return ModelTypes.TAR
+        else:
+            raise TrainerError(f"Model type {input_str} not supported.")
+
+    @staticmethod
+    def get_model_str(input_type):
+        """
+        Get the model configuration string from the model type.
+
+        Args:
+            input_type: input model enum.
+
+        Returns:
+            Model configuration string. "KIM", "TORCH" or "TAR"
+
+        """
+        if input_type == ModelTypes.KIM:
+            return "KIM"
+        elif input_type == ModelTypes.TORCH:
+            return "TORCH"
+        elif input_type == ModelTypes.TAR:
+            return "TAR"
+        else:
+            raise TrainerError(f"Model type {input_type} not supported.")
+
+
+class DataSource(Enum):
+    """
+    Enumerates the different types of data sources. The different types are:
+    - ASE: ASE atoms objects, or xyz file with configurations. Uses
+        ~:class:`~kliff.dataset.Dataset.from_ase` method.
+    - COLABFIT: uUses ColabFit dataset exchange instance. Uses
+        ~:class:`~kliff.dataset.Dataset.from_colabfit` method.
+    - KLIFF: Uses KLIFF compatible extxyz files path. Uses
+        ~:class:`~kliff.dataset.Dataset.from_path` method.
+    """
+
+    UNDEFINED = -1
+    ASE = 0
+    COLABFIT = 1
+    KLIFF = 2
```

Comment:
KLIFF can be confusing here. From a user's perspective, everything is kliff. We need to be more specific, like "EXTXYZ"?

Reply:
Removed enumerations, they were clunky and quite tedious to maintain. Will directly compare strings for now. "kliff" is renamed to path ("folder" might be better choice, but it is trivial to change)